src/Components/AddEditTask/TaskForm.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import { useNavigate, useLocation } from "react-router-dom";
+import { Task, TaskStatus , FormValues } from "../Types/type";
+import { useTasks } from "../Context/TaskProvider";
+import "./taskform.css";
+
+const schema = yup.object().shape({
+  title: yup
+    .string()
+    .required("Title is required")
+    .min(3, "Title must be at least 3 characters"),
+  desc: yup
+    .string()
+    .required("Description is required")
+    .min(5, "Description must be at least 5 characters"),
+});
+
+const TaskForm: ()  => React.ReactNode = () => {
+  const { tasks, setTasks } = useTasks();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const editId = location.state?.editId as string | undefined;
+  const taskToEdit = tasks.find((task) => task.id === editId);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<FormValues>({
+    resolver: yupResolver(schema),
+  });
+
+  useEffect(() => {
+    if (editId && taskToEdit) {
+      reset({
+        title: taskToEdit.title,
+        desc: taskToEdit.desc,
+      });
+    }
+  }, [editId, taskToEdit, reset]);
+
+  const onSubmit = (data: FormValues) => {
```

Comment:
The entire submit logic needs to be refactored.

```suggestion
  const onSubmit = (data: FormValues) => {
  let allTasks = [...tasks];
    if (taskToEdit) {
      allTasks[editId] = { ...data };
    } else {
      const id = crypto.randomUUID();
      allTasks = [...allTasks, { ...data, id }];
    }
    setTasks(allTasks);
    navigate("/");
```